File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerIdentityService.scala
// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.api.benchtool

import io.grpc.Channel
import com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc
import com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest
import org.slf4j.LoggerFactory

final class LedgerIdentityService(channel: Channel) {
  private val logger = LoggerFactory.getLogger(getClass)
  private val service = LedgerIdentityServiceGrpc.blockingStub(channel)

  def fetchLedgerId(): String = {
    val response = service.getLedgerIdentity(
      new GetLedgerIdentityRequest()
    )
    logger.info(s"Fetched ledger ID: ${response.ledgerId}")
    response.ledgerId
  }

}

File: sdk/language-support/hs/bindings/src/DA/Ledger/Services/LedgerIdentityService.hs
-- Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-- SPDX-License-Identifier: Apache-2.0

module DA.Ledger.Services.LedgerIdentityService (getLedgerIdentity) where

import Com.Daml.Ledger.Api.V1.LedgerIdentityService
import DA.Ledger.GrpcWrapUtils
import DA.Ledger.LedgerService
import DA.Ledger.Types(LedgerId(..))
import Network.GRPC.HighLevel.Generated

getLedgerIdentity :: LedgerService LedgerId
getLedgerIdentity =
    makeLedgerService $ \timeout config mdm -> do
    let request = GetLedgerIdentityRequest
    withGRPCClient config $ \client -> do
        service <- ledgerIdentityServiceClient client
        let LedgerIdentityService{ledgerIdentityServiceGetLedgerIdentity=rpc} = service
        response <- rpc (ClientNormalRequest request timeout mdm)
        GetLedgerIdentityResponse text <- unwrap response
        return $ LedgerId text
